rule/redundant_old_build_tag.go

```diff
@@ -0,0 +1,41 @@
+package rule
+
+import (
+	"strings"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// RedundantOldBuildTagRule lints the presence of old build tags `// +build`.
+type RedundantOldBuildTagRule struct{}
+
+// Apply triggers if an old build tag `// +build` is found after a new one `//go:build`.
+// `//go:build` comments are automatically added by gofmt when Go 1.17+ is used.
+// See https://pkg.go.dev/cmd/go#hdr-Build_constraints
+func (*RedundantOldBuildTagRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
+	var failures []lint.Failure
+
+	for _, group := range file.AST.Comments {
+		hasGoBuild := false
+		for _, comment := range group.List {
+			if strings.HasPrefix(comment.Text, "//go:build ") {
+				hasGoBuild = true
+			}
+			if hasGoBuild && strings.HasPrefix(comment.Text, "// +build ") {
+				failures = append(failures, lint.Failure{
+					Category:   "style",
+					Confidence: 1,
+					Node:       comment,
+					Failure:    `The old build tag "// +build" is redundant and can be removed`,
+				})
+			}
```

Comment:
`Apply` can return just after finding a failure. The probability of finding another `// +build` is very low